Bluetooth: controller: update remaining llcp unittests for new ZTEST API

kruithofa opened the pull request:
PR #52894, where the Feature Exchange unittests were updated, is the basis for this PR.

This PR updates the remaining LLCP unittests so that they are in line with the new ZTEST API.

Folowing are updated:
- API
- collision handling
- invalid PDUs
- unsupported procedures
- buffer allocation
- TX queue
- Data Length Update
- PHY update
- Encryption
- Ping
- Version
- Channel map update
- Min. used channels
- Connection update
- SCA
- Terminate connection
- CTE request
- CIS create
- CIS terminate

Also moved the internal API tests from the ull_llcp* files
to the proper unittest C-file

Unused files  are removed

NOTE: set to DNM until @tejlmand has had time to review

Commit:
bluetooth: controller: update unittests for LLCP to use proper ZTEST API

Update unittests for the following areas/controll procedures
- API
- collision handling
- invalid PDUs
- unsupported procedures
- buffer allocation
- TX queue
- Data Length Update
- PHY update
- Encryption
- Ping
- Version
- Channel map update
- Min. used channels
- Connection update
- SCA
- Terminate connection
- CTE request
- CIS create
- CIS terminate

Also moved the internal API tests from the ull_llcp* files
to the proper unittest C-file

Unused files  are removed

Signed-off-by: Andries Kruithof <andries.kruithof@nordicsemi.no>

erbr-ot commented on tests/bluetooth/controller/ctrl_chmu/prj.conf:
```diff
@@ -0,0 +1,30 @@
+CONFIG_ZTEST=y
```
Question - is it not possible to make a construct similar to -DOVERLAY_CONFIG  so there can be a common conf-file shared between tests, plus individual overlays?

kruithofa replied:
it would be possible yes, it is more or less what we did with the kconfig include-file hack. Now we have the unittest board it seems a bit more logical to have a prj.conf file for each 'project'. This can be discussed, and changed in a separate PR if we want to

erbr-ot reviewed (dismissed):
A couple of minor issues - nothing too important though. 
Looks pretty good

erbr-ot commented on subsys/bluetooth/controller/ll_sw/ull_llcp.c:
```diff
@@ -1849,142 +1849,13 @@ uint8_t common_tx_buffer_alloc_count(void)
 #endif /* LLCP_TX_CTRL_BUF_QUEUE_ENABLE */
 
-void test_int_mem_proc_ctx(void)
+struct proc_ctx *pub_proc_ctx_acquire(void)
```
oh, there was this one - why the 'pub' prefix here and the other places?

kruithofa replied:
so that we can expose some of the internal API for unit testing. pub is short for public

kruithofa replied:
and replaced the prefix with llcp_

kruithofa commented on tests/bluetooth/controller/ctrl_le_ping/src/main.c:
```diff
@@ -75,7 +74,7 @@ static void setup(void)
  *    |                            |                   |
  */
-void test_ping_central_loc(void)
+ZTEST(ping_central, test_ping_central_loc)
```
In the above example ping_central is the testgroup (i.e. all tests that do unit test for the ping procedure for the central), test_ping_central_loc is a particular unittest.

kruithofa commented:
rebased
@tejlmand can you review?

mbolivar-nordic commented on subsys/bluetooth/controller/ll_sw/ull_llcp.c:
```diff
@@ -1829,12 +1829,12 @@ void ull_cp_rx(struct ll_conn *conn, struct node_rx_pdu *rx)
 #ifdef ZTEST_UNITTEST
 
-static uint16_t local_ctx_buffers_free(void)
+uint16_t local_ctx_buffers_free(void)
```
The names of the functions in this file look quite generic to be removing `static`. Did you consider adding a prefix and renaming to something like `ull_local_ctx_buffers_free()`?

kruithofa replied:
Good point, even though they are only used for unittest of the module. I'll add the prefix.

Commit:
tests: bluetooth: add prefix to exposed API

In ull_llcp* some functions used in unittesting did not have
the llcp_ prefix
Although this is not a big issue since they are only used when
testing the module this commit adds the prefix so that we
have a more consistent naming

Signed-off-by: Andries Kruithof <andries.kruithof@nordicsemi.no>

tejlmand reviewed (approved):
lgtm. a minor comment, but not blocking.
